src/index.js

```diff
@@ -1,143 +1,270 @@
+            });
+    });
+};
 
 
 // Get the contents of the elm-package.json of the given package and reference
-var getPackageJson = function(package, ref){
-  var packageUrl = 'https://github.com/' + package + '/raw/' + ref + '/elm-package.json'
+var getPackageJson = function (packageName, ref) {
+
+    // console.log("getPackageJson: package=" + package + ", ref=" + ref);
+
+    if (isSsh(packageName)) {
+        return getPackageJsonSsh(packageName, ref);
+    } else {
+        var packageUrl = 'https://github.com/' + packageName + '/raw/' + ref + '/elm-package.json';
+        return new Promise(function (fulfill, reject) {
+            request.get(packageUrl, function (error, response, body) {
+                if (error) {
+                    reject(error);
+                } else {
+                    fulfill(JSON.parse(body));
+                }
+            });
+        });
+    }
+};
+
+var getPackageJsonSsh = function(packageName, ref) {
+
+    // console.log("getPackageJsonSsh: urlAndPackage=" + JSON.stringify(urlAndPackage, null, "  ") + ", ref=" + ref);
+
+    return new Promise(function (fulfill, reject) {
+
+        var repoName = packageName.split('/')[1];
+
+        var host = repoHost(packageName);
+
+        var cmd = 'cd /tmp && rm -rf ' + repoName +
+            ' && git clone --no-checkout --depth 1 --branch ' +
+            ref + ' git@' + host +
+            ':' + packageName + '.git &&' +
+            'cd ' + repoName + ' && git show HEAD:elm-package.json';
+
+        // console.log("getVersions: cmd=" + cmd);
+
+        exec(cmd, function(error, stdout, stderr){
+            if (error) {
+                console.error(stdout);
+                console.error(stderr);
+                reject(error);
+            } else {
+                fulfill(JSON.parse(stdout));
+            }
+        });
+
+    });
+};
 
-  return new Promise(function (fulfill, reject){
-    request.get(packageUrl, function(error, response, body){
-      fulfill(JSON.parse(body))
-    })
-  })
-}
 
 // Get all available versions (tags) for a given package
-var getVersions = function(package){
-  return new Promise(function (fulfill, reject){
-    var cmd = 'git ls-remote git://github.com/' +  package + ".git | awk -F/ '{ print $3 }'"
-    exec(cmd, function(error, stdout, stderr){
-      var versions = stdout.trim()
-                           .split("\n")
-                           .filter(function(version) {
-                              // filter out not valid tags (0.2.3^{})
-                              return semver.valid(version)
-                           })
-      fulfill(versions)
-    })
-  })
-}
+var getVersions = function (packageName) {
+
+    // console.log("getVersions: package=" + package);
+
+    return new Promise(function (fulfill, reject) {
+
+        var cmd;
+        if (isSsh(packageName)) {
+            cmd = 'git ls-remote git+ssh://' + repoHost(packageName) + '/' + packageName + ".git | awk -F/ '{ print $3 }'";
```

Comment:
You are right : I have tested `git ls-remote` on various repos (private and public, github, ghe and bitbucket) and it works. I have pushed a fix.